src/OpenLoco/src/Objects/AirportObject.cpp

```diff
@@ -127,4 +127,15 @@ namespace OpenLoco
         }
         return std::make_pair(minPos, maxPos);
     }
+
+    std::span<const AirportBuilding> AirportObject::getBuildingParts() const
+    {
+        const auto* firstPartPtr = buildingParts;
+        auto* endPartPtr = firstPartPtr;
+
+        while (endPartPtr->index != 0xFF)
+            endPartPtr++;
```

Comment:
I think you'll need to increment it by 4 each time

Reply:
Hmm, that doesn't seem to fix it. Surely it's incrementing by `sizeof(AirportBuilding)`, anyway?

Reply:
> Hmm, that doesn't seem to fix it. Surely it's incrementing by `sizeof(AirportBuilding)`, anyway?

oh right sorry didn't realise you changed the pointer type.